vnext/ReactUWP/Threading/ProxyMessageQueueThread.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <ReactWindowsCore/BatchingMessageQueueThread.h>
+#include <cxxreact/MessageQueueThread.h>
+
+namespace react {
+namespace uwp {
+
+/// This is a Message Queue Thread that wraps two seperate Message Queue
+/// Threads. The idea is to allow starting a react instance before a Window or
+/// UI Dispatcher are available (running background tasks) Then once
+/// AttachRootView is called a new Message Queue Thread can be provided with
+/// setUIMessageQueue.
+
+class ProxyMessageQueueThread
```

Comment:
>ProxyMessageQueueThread [](start = 6, length = 23)

It would be nice if we can use an alternative solution instead of implementing such proxy.

The multithreading development is already complex, and changing thread  somewhere in the middle sounds quite dangerous. It makes it difficult to reason if current work item is executed in a background or UI thread and if UI thread is required, then it will cause crashes.

I am going to bring to RNW the ReactNativeHost implementation from Office that must help us avoid 'engineering workarounds' like this.
In that model the ReactNativeHost is always in a background thread and only ReactNativeViewHost are the components that switch to UI thread.

Reply:
awesome thank you @vmoroz glad to hear there is already a better solution than this.